engine/classes/Elgg/HooksRegistrationService.php

```diff
@@ -56,6 +56,10 @@ public function registerHandler($name, $type, $callback, $priority = 500) {
 		if (empty($name) || empty($type) || !is_callable($callback, true)) {
 			return false;
 		}
+		
+		if (($name == 'view' || $name == 'view_vars') && $type != 'all') {
+			$type = _elgg_services()->views->canonicalizeViewName($type);
```

Comment:
I think this will break lots of handlers. Many I've written expect particular view names passed in.